src/libmongoc/src/mongoc/mongoc-client-session.c

```diff
@@ -1055,6 +1124,15 @@ mongoc_client_session_start_transaction (mongoc_client_session_t *session,
       GOTO (done);
    }
 
+   if (mongoc_session_opts_get_snapshot (&session->opts)) {
+      bson_set_error (error,
+                      MONGOC_ERROR_TRANSACTION,
+                      MONGOC_ERROR_TRANSACTION_INVALID_STATE,
+                      "Transactions are not supported in snapshot sessions");
+      ret = false;
+      GOTO (done);
```

Comment:
This new error is an [alteration](https://github.com/mongodb/specifications/pull/1022/files#diff-6c9d82f6b269142ff8fe9f0ef8d7c90f9e2761c947b5caf92e6820124b81fc2bR140) to the spec accompanied by a new spec test "StartTransaction fails in snapshot session".